sql/src/main/java/io/crate/metadata/cluster/ExchangeTableNameClusterStateExecutor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.cluster;
+
+import io.crate.execution.ddl.tables.RenameTableRequest;
+import io.crate.metadata.RelationName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.MetaDataIndexAliasesService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+
+public class ExchangeTableNameClusterStateExecutor extends AbstractRenameTableClusterStateExecutor {
+
+    public ExchangeTableNameClusterStateExecutor(Settings settings,
+                                                 IndexNameExpressionResolver indexNameExpressionResolver,
+                                                 MetaDataIndexAliasesService metaDataIndexAliasesService,
+                                                 NamedXContentRegistry namedXContentRegistry,
+                                                 DDLClusterStateService ddlClusterStateService) {
+        super(settings, indexNameExpressionResolver, metaDataIndexAliasesService,
+            namedXContentRegistry, ddlClusterStateService);
+    }
+
+    @Override
+    protected ClusterState execute(ClusterState currentState, RenameTableRequest request) throws Exception {
+        /*
+         x <-> y
+         :
+         y -> backup
+         x -> y
+         backup -> x
+         */
+
+        final boolean isPartitioned = request.isPartitioned();
+        final boolean validateClosed = false;
+        RelationName xRelationName = request.sourceTableIdent();
+        RelationName yRelationName = request.targetTableIdent();
+        RelationName backupRelationName = RelationName.fromIndexName(yRelationName.indexName() + "_backup");
+        ClusterState state = currentState;
+
+        // y -> backup
+        state = doExecute(state, yRelationName, backupRelationName, isPartitioned, validateClosed);
```

Comment:
imho we must operate on index names here, the operation should not have to know about partitions...

Reply:
done